### PR TITLE
feat: Added trackers for unified card enrollment flow

### PR DESCRIPTION
**Pull request by OmkarJ13**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=m_IeSbm)
### <samp>🤖 Generated by Copilot at 327ea86</samp>

This pull request adds tracking functionality for corporate card enrollment and unenrollment events in the mobile app. It modifies the `TrackingService` class to include new methods for tracking these events, and updates the `AddCorporateCardComponent` and `ManageCorporateCardsPage` classes to use them. It also adds unit tests for the tracking service calls in these components.

### <samp>🤖 Generated by Copilot at 327ea86</samp>

> _Tracking card events_
> _`TrackingService` helps us_
> _Autumn of errors_

### <samp>🤖 Generated by Copilot at 327ea86</samp>

*  Add methods to track card enrollment, unenrollment and enrollment failure events in `TrackingService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-63225a1ea9f44d74739fc2787c222f8a94585abcf21a01bf159adbb4c6597d7aR612-R623))
*  Inject and mock `TrackingService` and `Router` modules in unit tests for `AddCorporateCardComponent` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR15-R16), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR34-R35), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR44), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR59-R68), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR77-R78))
*  Add unit test cases to verify tracking service calls with correct properties on card enrollment success and failure ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR299-R327), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-c5548f8968bea9df856f1bf7ad8b68d9b7eee71d7181ef5cf14b83c59904b45dR380-R408))
*  Import and inject `TrackingService` and `Router` modules in `AddCorporateCardComponent` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-85cca3686bdc76c7d1757fed2d3cfd9e2c68755b19b4d0edc64a6fab2b1c771dR3), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-85cca3686bdc76c7d1757fed2d3cfd9e2c68755b19b4d0edc64a6fab2b1c771dR9), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-85cca3686bdc76c7d1757fed2d3cfd9e2c68755b19b4d0edc64a6fab2b1c771dR42-R43))
*  Modify `handleEnrollmentSuccess` and `handleEnrollmentFailure` methods to call tracking service methods with card information and source URL ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-85cca3686bdc76c7d1757fed2d3cfd9e2c68755b19b4d0edc64a6fab2b1c771dL82-R86), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-85cca3686bdc76c7d1757fed2d3cfd9e2c68755b19b4d0edc64a6fab2b1c771dL152-R174))
*  Inject and mock `TrackingService` module in unit tests for `ManageCorporateCardsPage` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R34), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R59), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R72), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R106-R109), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8R125))
*  Modify unit test case to verify tracking service call with correct properties on card unenrollment ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8L341-R349))
*  Remove redundant assertions for popover creation and presentation in unit test case for card unenrollment ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-b389a5d0241bda35125b36240b1d1cc470932c179782063153e1d1ad67705ed8L312-L329))
*  Import and inject `TrackingService` module in `ManageCorporateCardsPage` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fR17), [link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fR43))
*  Add call to tracking service method with card information in `disconnectCard` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2399/files?diff=unified&w=0#diff-450aa28d32fd44fd50bd49190bd3028ca78d9b54b8849b79192b7a2fc10a008fR208-R212))

## Clickup
https://app.clickup.com

### CODE REVIEW

File: src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.spec.ts [hunk 1]
```diff
@@ -338,28 +328,25 @@ describe('ManageCorporateCardsPage', () => {
 
       tick();
 
-      expect(popoverController.create).toHaveBeenCalledOnceWith({
```
Comment (OmkarJ13) [inline src/app/fyle/manage-corporate-cards/manage-corporate-cards.page.spec.ts:341]: Removed unnecessary expects in other tests

File: src/app/fyle/manage-corporate-cards/add-corporate-card/add-corporate-card.component.spec.ts [hunk 2]
```diff
@@ -154,7 +174,15 @@ describe('AddCorporateCardComponent', () => {
       fixture.detectChanges();
 
       const errorMessage = getElementBySelector(fixture, '[data-testid="error-message"]') as HTMLElement;
+
       expect(errorMessage.innerText).toBe('Please enter a valid card number.');
+      expect(trackingService.cardEnrollmentErrors).toHaveBeenCalledOnceWith({
```
Comment (Dimple16) [inline src/app/fyle/manage-corporate-cards/add-corporate-card/add-corporate-card.component.spec.ts:179]: can we add these to tracking-properties model? it'll avoid repetition

**Review by Dimple16 — CHANGES_REQUESTED** [review]
one comment

File: src/app/fyle/manage-corporate-cards/add-corporate-card/add-corporate-card.component.spec.ts (hunk 2 above)
Comment (OmkarJ13) [reply src/app/fyle/manage-corporate-cards/add-corporate-card/add-corporate-card.component.spec.ts]: Have added the tracking properties to mock data and added models for each event

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**